--- ext-3rd/elfinder/js/elfinder.full.js
+++ ext-3rd/elfinder/js/elfinder.full.js
@@ -1525,7 +1525,7 @@
 					position    : p,
 					title       : self.fm.i18n(f.mime == 'directory' ? 'Folder info' : 'File info'),
 					close       : function() { if (--cnt <= 0) { self.fm.lockShortcuts(); } $(this).dialog('destroy'); },
-					buttons     : { Ok : function() { $(this).dialog('close'); }}
+					buttons     : [ { text : self.fm.i18n('Close'), click: function() { $(this).dialog('close'); }}]
 				});
 			}
 		}
@@ -1754,9 +1754,9 @@
 						dialogClass : 'el-finder-dialog',
 						width       : 350,
 						close       : function() { self.fm.lockShortcuts(); },
-						buttons     : {
-							Cancel : function() { $(this).dialog('close'); },
-							Ok     : function() { 
+						buttons     : [
+							{ text : self.fm.i18n('Cancel'), click : function() { $(this).dialog('close'); }},
+							{ text : self.fm.i18n('Remove'), click : function() { 
 								$(this).dialog('close'); 
 								var o = { cmd : 'rm', current : self.fm.cwd.hash };
 								if (self.fm.jquery > 132) {
@@ -1765,8 +1765,8 @@
 									o['targets[]'] = ids;
 								}
 								self.fm.ajax(o, function(data) { data.tree && self.fm.reload(data); }, {force : true});
-							}
-						}
+							}}
+						]
 					});
 			}
 		}
@@ -1963,9 +1963,9 @@
 						modal       : true,
 						resizable   : false,
 						close       : function() { self.fm.lockShortcuts(); },
-						buttons     : {
-							Cancel : function() { $(this).dialog('close'); },
-							Ok     : function() {
+						buttons     : [
+							{ text : self.fm.i18n("Cancel"), click : function() { $(this).dialog('close'); }},
+							{ text : self.fm.i18n("Upload"), click : function() {
 								if (!$(':file[value]', f).length) {
 									return error(self.fm.i18n('Select at least one file to upload'));
 								}
@@ -1983,8 +1983,8 @@
 									}
 								});
 								$(this).dialog('close');
-							}
-						}
+							}}
+						]
 					});
 
 			self.fm.lockShortcuts(true);
@@ -2115,9 +2115,9 @@
 						modal   : true,
 						width   : 500,
 						close   : function() { self.fm.lockShortcuts(); },
-						buttons : {
-							Cancel : function() { $(this).dialog('close'); },
-							Ok     : function() {
+						buttons : [
+							{ text : self.fm.i18n('Cancel'), click : function() { $(this).dialog('close'); }},
+							{ text : self.fm.i18n('Save'), click   : function() {
 								var c = ta.val();
 								$(this).dialog('close');
 								self.fm.ajax({
@@ -2133,8 +2133,8 @@
 										
 									}
 								}, {type : 'POST'});
-							}
-						}
+							}}
+						]
 					});
 			});
 		}
