0000832: Ошибка при наличии разных префиксов в БД
URL: http://bugs.eresus.ru/view.php?id=832
===================================================================
--- core/framework/core/3rdparty/ezcomponents/DatabaseSchema/src/handlers/mysql/reader.php
+++ core/framework/core/3rdparty/ezcomponents/DatabaseSchema/src/handlers/mysql/reader.php
@@ -68,7 +68,16 @@
      */
     protected function fetchSchema()
     {
-        $tables = $this->db->query( "SHOW TABLES" )->fetchAll();
+        $prefix = ezcDbSchema::$options->tableNamePrefix;
+        if (empty($prefix))
+        {
+            $sql = "SHOW TABLES";
+        }
+        else
+        {
+            $sql = "SHOW TABLES LIKE '$prefix%'";
+        }
+        $tables = $this->db->query( $sql )->fetchAll();
         return $this->processSchema( $tables );
     }
 
